app/src/main/java/org/p2p/wallet/home/MainFragment.kt

```diff
@@ -65,6 +71,9 @@ class MainFragment : BaseFragment(R.layout.fragment_main), MainTabsSwitcher, Cen
 
     private var lastSelectedItemId = R.id.homeItem
 
+    private val mainScreenActionButtonHandler: MainScreenActionButtonHandler?
+        get() = tabCachedFragments[R.id.homeItem] as? MainScreenActionButtonHandler
```

Comment:
Why did you add a listener here, not directly in HomeFragment?

Reply:
according to what we have there is no way no do it
we need to call function from historyFragment to homeFragment 
this was fast way

another way is to use DI or re write code and make calls to buy and receive in MainFragment 

or add same code in history fragment

also we can use deeplinks in app

but all this consumes time and I think must be done in separate task

what do you think ?